PurchasesCoreSwift/Identity/IdentityManager.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  IdentityManager.swift
+//
+//  Created by Joshua Liebowitz on 8/9/21.
+
+import Foundation
+
+// TODO (post-migration): Change back to internal, including all public properties.
+@objc(RCIdentityManager) public class IdentityManager: NSObject {
+
+    @objc public var currentAppUserID: String? {
+        return deviceCache.cachedAppUserID
+    }
+
+    @objc public var currentUserIsAnonymous: Bool {
+        let anonymousRegex = #"\$RCAnonymousID:([a-z0-9]{32})$"#
+        let anonymousFoundRange = currentAppUserID?.range(of: anonymousRegex, options: .regularExpression)
+        let foundAnonymous = anonymousFoundRange != nil && !(anonymousFoundRange?.isEmpty ?? false)
+        let currentAppUserIDLooksAnonymous = foundAnonymous
+
+        let isLegacyAnonymousAppUserID: Bool
+        if let cachedAppUserID = currentAppUserID {
+            isLegacyAnonymousAppUserID = cachedAppUserID == deviceCache.cachedLegacyAppUserID
+        } else {
+            isLegacyAnonymousAppUserID = false
+        }
+
+        return currentAppUserIDLooksAnonymous || isLegacyAnonymousAppUserID
+    }
+
+    private let deviceCache: DeviceCache
+    private let backend: Backend
+    private let purchaserInfoManager: PurchaserInfoManager
+
+    @objc public init(deviceCache: DeviceCache, backend: Backend, purchaserInfoManager: PurchaserInfoManager) {
+        self.deviceCache = deviceCache
+        self.backend = backend
+        self.purchaserInfoManager = purchaserInfoManager
+    }
+
+    @objc public func configure(appUserID maybeAppUserID: String?) {
+        let appUserID = maybeAppUserID
+            ?? deviceCache.cachedAppUserID
+            ?? deviceCache.cachedLegacyAppUserID
+            ?? generateRandomID()
+        Logger.user(String(format: Strings.identity.identifying_app_user_id, appUserID))
+
+        deviceCache.cache(appUserID: appUserID)
+        deviceCache.cleanupSubscriberAttributes()
+    }
+
+    @objc public func logIn(appUserID: String, completion: @escaping (PurchaserInfo?, Bool, Error?) -> Void) {
+        let newAppUserID = appUserID.trimmingCharacters(in: .whitespacesAndNewlines)
+        guard let currentAppUserID = currentAppUserID,
```

Comment:
since we can't change `currentAppUserID` to `maybeCurrentAppUserID`, can we somehow else distinguish these two? maybe `guard let nonNullCurrentAppUserId`?

Reply:
I think you bring up a good point, and I would like to change the name of the variable. This isn't a public API for our users, so we CAN change it `maybeCurrentAppUserID` 🎉